cogs/vxtwitconverter/vxtwitconverter.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import os
+
+from discord import Message, channel
+from urlextract import URLExtract
+
+from redbot.core import commands, data_manager
+from redbot.core.bot import Red
+
+
+class VxTwitConverter(commands.Cog):
+    """Converts twitter link embeds to vxtwitter"""
+
+    def __init__(self, bot: Red):
+        super().__init__()
+        self.bot = bot
+
+        # Initialize logger, and save to cog folder.
+        saveFolder = data_manager.cog_data_path(cog_instance=self)
+        self.logger = logging.getLogger("red.luicogs.vxtwitconverter")
+        if not self.logger.handlers:
+            logPath = os.path.join(saveFolder, "info.log")
+            handler = logging.FileHandler(filename=logPath, encoding="utf-8", mode="a")
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
+            )
+            self.logger.addHandler(handler)
+
+    @commands.Cog.listener("on_message")
+    async def twit_replacer(self, message: Message):
+        # skips if the message is sent by any bot
+        if message.author.bot:
+            return
+
+        # skips if message is in dm
+        if isinstance(message.channel, channel.DMChannel):
+            return
+
+        # skips if the message has no embeds
+        if not any(embed.video for embed in message.embeds):
+            return
+
+        # the actual code part
+        extractor = URLExtract()
+
+        if extractor.has_urls(message.content):
+            results = extractor.find_urls(message.content)
+            new_message = [
+                result.replace("https://twitter.com", "https://vxtwitter.com")
+                for result in results
+                if "https://twitter.com" in result
+            ]
+
+            if new_message:
+                # removed embed from the parent message, and replies with vxtwitter link
+                await message.reply(
+                    "OwO what's this?\n"
+                    "*notices your terrible twitter embeds*\n"
+                    "Here's a better alternative:\n" + ",\n".join(new_message)
+                )
+
+                await message.edit(suppress=True)
```

Comment:
this is why I have suppress after reply, so if reply throws an error; the original message doesn't get suppressed since the code will just stop excecution

Reply:
Oh actually because `new_message` here contains only VxTwitter links so that's fine. When I wrote this comment, I thought `new_message` was to contain the original `message` with Twitter links replaced. So if a message contains only links, I don't expect `message.reply` to fail ever (excluding transient HTTP errors we get from time to time with Discord, which can cause an exception, but we should be able to recognize it via logs).

That said, `new_message` somewhat implies a meaning of an updated version of the variable `message`. Perhaps something like `vxTwitMessage` can be a better name? 🤔 

Regarding the point of try-except, it's needed to gracefully handle errors (if any) so I'm not sure how you go about it normally. In this case I don't think `message.reply` fails in most cases so we don't need.